stuffer/s2n_stuffer_network_order.c

```diff
@@ -38,7 +38,7 @@ int s2n_stuffer_write_network_order(struct s2n_stuffer *stuffer, const uint64_t
 
 int s2n_stuffer_reserve(struct s2n_stuffer *stuffer, struct s2n_stuffer_reservation *reservation, uint8_t length)
 {
-    notnull_check(stuffer);
+    PRECONDITION_POSIX(s2n_stuffer_is_valid(stuffer));
     notnull_check(reservation);
 
     reservation->stuffer = stuffer;
```

Comment:
this is a style thing, but I prefer explicit initilization: 

```
*reservation = (struct s2n_stuffer_reservation) {.stuffer = stuffer, .write_cursor = stuffer->write_cursor, .length =length};
```

Reply:
Doesn't that have an issue if a new field is added?

Reply:
"Omitted field members are implicitly initialized the same as objects that have static storage duration." (https://gcc.gnu.org/onlinedocs/gcc/Designated-Inits.html)

Reply:
Right, that's what I was trying to say. If a field is added, the stuffer copy wouldn't be correct and the compiler wouldn't tell us either.

Reply:
Reading through the code more, I think @danielsn is right here. The caller most likely hasn't initialized the fields (see https://github.com/awslabs/s2n/blob/603f4f0b3d35eccc0abc94765122a38974389585/tests/unit/s2n_extension_list_parse_test.c#L190 as an example)